android/engine/src/main/java/org/smartregister/fhircore/engine/util/extension/QuestionnaireExtension.kt

```diff
@@ -146,8 +146,9 @@ fun List<Questionnaire.QuestionnaireItemComponent>.prePopulateInitialValues(
   prePopulationParams: List<ActionParameter>
 ) {
   forEach { item ->
-    if (item.hasExtension(ITEM_INITIAL_EXPRESSION_URL))
+    if (item.hasExtension(ITEM_INITIAL_EXPRESSION_URL)) {
       item.removeExtension(ITEM_INITIAL_EXPRESSION_URL)
+    }
     prePopulationParams
```

Comment:
```
 prePopulationParams
      .firstOrNull {
        it.linkId == item.linkId &&
          !it.value.isNullOrEmpty() &&
          !it.value.contains(interpolationPrefix)
      }
      ?.let { actionParam ->

 if (item.hasExtension(ITEM_INITIAL_EXPRESSION_URL)) {
      item.removeExtension(ITEM_INITIAL_EXPRESSION_URL)
    }

        item.initial =
          arrayListOf(
            Questionnaire.QuestionnaireItemInitialComponent().apply {
              value = actionParam.dataType?.let { actionParam.value.castToType(it) }
            }
          )
      }
```
      
      wouldn't check `removeExtension` go inside of this scope once prepopulaedPrams linkId matches with one of the questionnaire item's linkId ? 
      
      @FikriMilano

Reply:
wouldn't check `removeExtension` go inside of this scope once prepopulaedPrams linkId matches with one of the questionnaire item's linkId ? @FikriMilano

Reply:
yes that makes sense, that is unrelated to this PRs specific topic, @aurangzaibumer very good catch, I've opened an issue for this here https://github.com/opensrp/fhircore/issues/2335, please add any additional context or comment

Reply:
@aurangzaibumer thanks, nice catch! 
Please do add the change